framework/framework.go

```diff
@@ -142,3 +142,60 @@ func (f *Framework) ProcessDelete(obj interface{}, resources []Resource) error {
 
 	return nil
 }
+
+// ProcessUpdate is a drop-in for an informer's UpdateFunc. It receives the
+// custom object observed during TPR watches and anything that implements
+// Resource. ProcessUpdate takes care about all necessary reconciliation logic
+// for update events. For complex resources this means state has to be created,
+// deleted and updated eventually, in this order.
+//
+//     func updateFunc(oldObj, newObj interface{}) {
+//         err := f.ProcessUpdate(oldObj, newObj, resources)
+//         if err != nil {
+//             // error handling here
+//         }
+//     }
+//
+//     newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
+//         UpdateFunc:    updateFunc,
+//     }
+//
+func (f *Framework) ProcessUpdate(oldObj, newObj interface{}, resources []Resource) error {
```

Comment:
`OnAdd` and `OnUpdate` are both called when a re-list happens. See https://godoc.org/k8s.io/client-go/tools/cache#ResourceEventHandler. `OnAdd` should only be concerned about adding new resources. `OnUpdate` should only be concerned about updating resources. Updating resource implies adding new resources, but this should only reuse functionality used by `OnAdd`. `ProcessCreate` as defined by the framework has to be there because this defines the primitive of adding new resources. No matter if it is invoked by `OnAdd` or `OnUpdate`. We could argue that it does not make sense to have `OnAdd` and `OnUpdate` because `OnUpdate` implies `OnAdd`, but this is not in the scope of the operatorkit reconciliation framework.